Patches/PlayerControlPatch.cs

```diff
@@ -2021,7 +2018,7 @@ public static bool Prefix(PlayerControl __instance, [HarmonyArgument(0)] GameDat
                         }
                         else if (tar.Is(CustomRoles.Admirer))
                         {
-                            Admirer.Add(__instance.PlayerId);
+                            CustomRoles.Admirer.GetRoleClass()?.Add(__instance.PlayerId);
```

Comment:
put this after the rpcrole has been set.
And get the object from Main.PlayerStates[tar.PlayerId].Role;